### PR TITLE
projects/Amlogic: add S912 support

### PR DESCRIPTION
**Pull request by kszaq**
This is a follow-up of https://github.com/LibreELEC/LibreELEC.tv/pull/2400 after @adamg88 decided to close his PR. Fortunately I had a copy of the branch so we can continue the work.

The main difference between already-supported S905X and S912 is that the latter has an additional CPU cluster (already covered by our kernel config) and Mali-T820 GPU that can be used via libhybris.

This PR includes a few device-specific packages that have no use in other projects, that are:
- **libhybris** with a patch to build only EGL libraries - everything else is supported natively, second libhybris patch was PRd https://github.com/libhybris/libhybris/pull/371
- **android-headers** to compile libhybris - these are extracted from AOSP and packaged
- **opengl-meson-t82x** - libMali.so from Android with its dependencies - extracted from Khadas VIM2 firmware

List of TODOs:
- [x] <del> figure out a way to resize EGL window when resolution changes - libhybris doesn't proivde fbdev_window - this would allow us to get rid of Kodi hack</del>
- [x] <del>compare GPU performance without DVFS and Turbo patches</del>
- [x] <del>consider merging **gpu-aml-t82x** with **gpu-aml**</del>

### CODE REVIEW

**Comment by kszaq** [comment]
1. Rebased on master.
2. Merged **gpu-aml-t8xx** with **gpu-aml** and got rid of patches.

"squashme" notes added to squash before finalizing.

File: projects/Amlogic/devices/S912/options [hunk 1]
```diff
@@ -0,0 +1,13 @@
+  # Kernel extra targets to build
+     KERNEL_UBOOT_EXTRA_TARGET="gxm_q200_2g.dtb gxm_q201_1g.dtb gxm_q201_2g.dtb"
+
+  # OpenGL-ES implementation to use (no / bcm2835-driver / gpu-viv-bin-mx6q / opengl-meson)
+     OPENGLES="opengl-meson-t82x"
```
Comment (adamg88) [inline projects/Amlogic/devices/S912/options]: Just a nitpick with the formatting, is there extra whitespaces here?

Comment (kszaq) [reply projects/Amlogic/devices/S912/options]: I will fix this when squashing, thanks.

**Comment by adamg88** [comment]
I think it's worth remembering exactly what JeOS stands for

**Comment by adamg88** [comment]
I know some of those drivers are not in S912 devices and should be excluded if they are not needed imo.

**Comment by kszaq** [comment]
This PR is now ready (for reviews).

Please note that to build Kodi you need to update it to https://github.com/xbmc/xbmc/commit/e4074d84d1809f7959e19df12ac3d23950f39a49 or newer.

File: projects/Amlogic/devices/S912/packages/android-headers/package.mk [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+#  along with LibreELEC.  If not, see <http://www.gnu.org/licenses/>.
+################################################################################
+
+PKG_NAME="android-headers"
+PKG_VERSION="25"
+PKG_SHA256="1e0ecdf56c33aaa523109254e2c475878d8cfc5795ebd4bb5ecbaf80926f4fe9"
+PKG_ARCH="any"
+PKG_LICENSE="Apache"
+PKG_SITE="https://android.googlesource.com/"
+PKG_URL="https://kszaq.libreelec.tv/sources/${PKG_NAME}-${PKG_VERSION}.tar.gz"
```
Comment (CvH) [inline projects/Amlogic/devices/S912/packages/android-headers/package.mk]: `PKG_URL="$DISTRO_SRC/$PKG_NAME-$PKG_VERSION.tar.gz"` :)

Comment (kszaq) [reply projects/Amlogic/devices/S912/packages/android-headers/package.mk]: Done.

**Comment by kszaq** [comment]
Merge blocked added as Utgard driver does not work properly after update.

**Comment by kszaq** [comment]
I consider this PR to be complete.

**Pull request merged**